### PR TITLE
router: 1.30.1 -> 1.49.1

### PR DESCRIPTION
**Pull request by bbigras**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/servers/http/router/default.nix [hunk 1]
```diff
@@ -35,15 +35,15 @@ rustPlatform.buildRustPackage rec {
   # To avoid this we pre-download the file and export it via RUSTY_V8_ARCHIVE
   RUSTY_V8_ARCHIVE = callPackage ./librusty_v8.nix { };
 
+  doCheck = false;
```
Comment (getchoo) [inline pkgs/servers/http/router/default.nix:38]: a comment could be nice about why all tests are disabled

**Review by getchoo — CHANGES_REQUESTED** [review]

File: pkgs/servers/http/router/default.nix (hunk 1 above)
Comment (bbigras) [reply pkgs/servers/http/router/default.nix]: my bad. It's removed.

File: pkgs/servers/http/router/default.nix [hunk 2]
```diff
@@ -10,16 +10,16 @@
 
 rustPlatform.buildRustPackage rec {
   pname = "router";
-  version = "1.30.1";
+  version = "1.45.1";
```
Comment (LeSuisse) [inline pkgs/servers/http/router/default.nix:13]: Is it possible to bump to 1.49.1?

https://github.com/apollographql/router/releases/tag/v1.49.1

File: pkgs/servers/http/router/default.nix [hunk 3]
```diff
@@ -10,16 +10,16 @@
...
 
   src = fetchFromGitHub {
     owner = "apollographql";
     repo = pname;
     rev = "v${version}";
-    sha256 = "sha256-mQtIjfXDcEy5HfZbWauL0NQLPneGq9EJt/yB8zMuhSU=";
+    sha256 = "sha256-GmSQyFVaCfx2keaFgqtI0iFUvYoH3KQzyQCRzE0D920=";
```
Comment (LeSuisse) [inline pkgs/servers/http/router/default.nix:19]:
```suggestion
    hash = "sha256-GmSQyFVaCfx2keaFgqtI0iFUvYoH3KQzyQCRzE0D920=";
```

File: pkgs/servers/http/router/default.nix (hunk 2 above)
Comment (bbigras) [reply pkgs/servers/http/router/default.nix:13]: Some tests were failing with 1.49.1, so I wanted to fix the vulnerability asap.

Comment (LeSuisse) [reply pkgs/servers/http/router/default.nix:13]: I made a quick a test with 1.49.1 with your current changes, it seems to build correctly?

Comment (bbigras) [reply pkgs/servers/http/router/default.nix:13]: Maybe the problem was only on my side. I'll try the latest and see if CI is fine with it.

**Comment by bbigras** [comment]
@ofborg build

**Comment by bbigras** [comment]
@ofborg build router

**Comment by bbigras** [comment]
@LeSuisse can run nixpkgs-review and post the result please?

**Comment by bbigras** [comment]
@LeSuisse I just did a force push.

**Review by LeSuisse — APPROVED** [review]
Looks good and it seems to work fine, let's merge.


Result of `nixpkgs-review pr 308415` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>router</li>
  </ul>
</details>

**Comment by bbigras** [comment]
Thanks!

**Pull request merged**